perf/perf.jl

```diff
@@ -0,0 +1,100 @@
+using Flux, GeometricFlux, LightGraphs, BenchmarkTools, CUDA
+using DataFrames, Statistics, JLD2, SparseArrays
+
+BenchmarkTools.ratio(::Missing, x) = Inf
+BenchmarkTools.ratio(x, ::Missing) = 0.0
+BenchmarkTools.ratio(::Missing, ::Missing) = missing
+
+adjlist(g) = [neighbors(g, i) for i in 1:nv(g)]
+
+function run_single_benchmark(N, K, D, CONV; gtype=:lg)
+    @assert gtype ∈ [:lightgraph, :adjlist, :dense, :sparse]
+    g = random_regular_graph(N, K, seed=17)
```

Comment:
In network science, an Erdős–Rényi random graph is usually used as baseline random graph generator model. Would you pleas change to [`erdos_renyi`](https://juliagraphs.org/LightGraphs.jl/v1.1/generators.html#LightGraphs.SimpleGraphs.erdos_renyi-Tuple{Integer,Real}) and give a probability `p` that represents the possibility of two nodes are connected.

Reply:
I think we can set the probability `p = 0.2`.

Reply:
`p` is commonly scaled with `N`. Emergence of the giant component is at the critical value `p = 1 / N` for N that goes to infinity. I could set `p = c / (N-1)` for a graph with average connectivity `c` and expose `c` as a function argument.

Reply:
Yes, it is naturally scaled with node number. Natural graph usually scales with node number, so it is natural to scaled with node number in order to mimic natural graph. It doesn't need to avoid the scaling issue.

Reply:
I mean just use `p` as usual. We don't need to benchmark the graph with node number increase but edge number fixed. Benchmark over a graph with fixed edge number is nonsense.

Reply:
> Benchmark over a graph with fixed edge number is nonsense.

There is some misunderstanding here. The number of edges in this PR scales as O(N). What you suggest is O(N^2) and as I already commented https://github.com/FluxML/GeometricFlux.jl/pull/214#issuecomment-892381241 is too dense

Reply:
OK. I see.